### ISSUE TITLE
Blank screen Ipad, Android

### ISSUE THREAD

**Issue by andwah** [Starcounter/Home#296]
Navigating an an app i get a blank screen on:

Chrome on Android phone
Chrome on Ipad
Safari on Ipad

Works fin with Chrome (Version 62.0.3202.75) on Windows.

Using Starcounter-RC-2.3.1.7581

This issue may be related https://github.com/Starcounter/Home/issues/285

**Comment by alshakero** [comment]
The reason seems to be ShadowDOMv1. Using SDv0 fixes the issue everywhere.

Set this to `true` and all works:
https://github.com/Starcounter/starcounter-include/blob/master/starcounter-include.html#L19

@warpech @tomalec any suggestions?

**Comment by warpech** [comment]
How are you testing this?

My results when testing KitchenSink in Starcounter 2.3.1.7680 (from https://github.com/Starcounter/level1/pull/4331) with no Shadow DOM related changes:

- Safari 11 on iOS 11.0.3 works
- Chrome 62 on iOS 11.0.3 ("Request mobile site") does **not** work for me
- Chrome 62 on iOS 11.0.3 ("Request desktop site") works
- Chrome 61 on Android 6.0.1 works

**Comment by alshakero** [comment]
I did exactly the same way! I downloaded this PR's build and got the same results. Disabling SDv1 fixes Chrome on iOS. And if you're asking technically, I debug:
- iOS Safari using Safari dev tools on Mac
- Chrome Android (I have an old Android phone I bought for testing) on Chrome Dev tools
- I don't have a way to debug Chrome iOS (there isn't).

**Comment by warpech** [comment]
Maybe this works for you https://github.com/RemoteDebug/remotedebug-ios-webkit-adapter

**Comment by alshakero** [comment]
@warpech this is for Safari. Safari iOS is debuggable enough using Safari MacOS.

**Comment by warpech** [comment]
I know the reason why it fails. 

In `starcounter-include` we have this check for Shadow DOM v1:

```
const isSafari = navigator.vendor && navigator.vendor.indexOf("Apple") > -1 && navigator.userAgent && !navigator.userAgent.match("CriOS");
const useShadowDOMV1 = !isSafari && Boolean(Element.prototype.attachShadow && Node.prototype.getRootNode);
```

So here's the deal:

- Chrome 62 on iOS 11.0.3 ("Request mobile site") has UA string `Mozilla/5.0 (iPhone; CPU iPhone OS 11_0_3 like Mac OS X) AppleWebKit/604.1.34 (KHTML, like Gecko) CriOS/62.0.3202.70 Mobile/15A432 Safari/604.1`. Existence of `CriOS` in this string causes `isSafari == false`, therefore `useShadowDOMV1 == true`
- Chrome 62 on iOS 11.0.3 ("Request desktop site") has UA string `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12`, which resolves to `isSafari == true`, therfore `useShadowDOMV1 == false`

We need to fix the `isSafari` to always be true for iOS web views.

**Comment by warpech** [comment]
https://github.com/Starcounter/starcounter-include/commit/a8908ebcaa4259f18be88fcdfb57a7e7d9e4709a fixes Chrome on iOS. 

It will be released as `starcounter-include` 3.0.0-rc.9 and added to https://github.com/Starcounter/level1/pull/4331

**Comment by warpech** [comment]
Closing as done
